src/components/ICA/__tests__/ICA.spec.js

```diff
@@ -3,76 +3,75 @@
  * @copyright IBM Security 2019
  */
 
-import { shallow } from 'enzyme';
 import React from 'react';
+import { render } from '@testing-library/react';
 
 import { ICA } from '../../..';
 import { Locales } from '../ICA';
 
-import props from '../_mocks_';
-
-const { label, total, value } = props;
-
-const defaultProps = { label, value };
-
 describe('ICA', () => {
-  let ica;
-  const largeValue = value * total;
-
-  beforeEach(() => {
-    ica = shallow(<ICA {...defaultProps} />);
+  test('should have no Axe or DAP violations', async () => {
+    const main = document.createElement('main');
+    render(<ICA label="test ICA" total={10} value={5} />, {
+      // DAP requires a landmark '<main>' in the DOM:
+      container: document.body.appendChild(main),
+    });
+    await expect(document.body).toHaveNoAxeViolations();
+    await expect(document.body).toHaveNoDAPViolations('ICA');
   });
 
-  it('renders', () => {
-    expect(ica).toMatchSnapshot();
+  test('should render en dash when `value` is `null`', () => {
+    const { queryByText } = render(<ICA label="test ICA" value={null} />);
+    expect(queryByText('–')).toBeVisible();
   });
 
-  it('renders the null value', () => {
-    ica.setProps({ value: null });
-
-    expect(ica).toMatchSnapshot();
+  test('should render a large value', () => {
+    const { queryByText } = render(<ICA label="test ICA" value={1000000} />);
+    expect(queryByText('1.0m')).toBeVisible();
   });
 
-  it('renders a large value', () => {
-    ica.setProps({ value: largeValue });
-
-    expect(ica).toMatchSnapshot();
+  test('should render a subset of values when `value` and `total` are provided', () => {
+    const { getByText } = render(<ICA label="test ICA" value={5} total={10} />);
+    expect(getByText('5').nextSibling).toBe(getByText(/10/i));
   });
 
-  it('renders a subset of values', () => {
-    ica.setProps({ total });
-
-    expect(ica).toMatchSnapshot();
+  test('should not render subset of values when `total` is close to `value`', () => {
+    const { queryByText } = render(
+      <ICA label="test ICA" value={1000000} total={999995} />
+    );
+    expect(queryByText('999995')).not.toBeInTheDocument();
+    expect(queryByText('1.0m')).toBeVisible();
   });
 
-  it('does not render a subset when the total is close to the provided value', () => {
-    ica.setProps({ total: largeValue - value, value: largeValue });
-    expect(ica).toMatchSnapshot();
+  test('should not render `total` if it is the same as `value`', () => {
+    const { getByText } = render(
+      <ICA label="test ICA" value={10} total={10} />
+    );
+    // Expect 2 ICA nodes: the label and the value.
+    expect(getByText(/test ICA/i).parentNode.children.length).toBe(2);
```

Comment:
Is it sufficient to use `queryByText(' / 10').not.toBeInTheDocument()` here?

Reply:
This would not produce the result you want, possibly because of the new line before/after the slash.